stack-graphs/src/partial.rs

```diff
@@ -2069,13 +2069,14 @@ impl PartialPath {
         graph[self.end_node].is_jump_to()
     }
 
-    /// Returns whether a partial path is "productive" --- that is, it is not cyclic, or its
-    /// pre- and postcondition are incompatible.
-    pub fn is_productive(&self, graph: &StackGraph, partials: &mut PartialPaths) -> bool {
+    /// Returns whether a partial path is cyclic---that is, it starts and ends at the same node,
+    /// and its postcondition is compatible with its precondition.  If the path is cyclic, a
+    /// tuple is returned indicating whether cycle requires strengthening the pre- or postcondition.
+    pub fn is_cyclic(&self, graph: &StackGraph, partials: &mut PartialPaths) -> Option<Cyclicity> {
```

Comment:
This would end up becoming just the bitset type, as well, since the bitset being empty would encode the `None` case

Reply:
This remains as it is: if the path is cyclic, it returns its character, otherwise none. But a single cycle cannot be cyclic in multiple ways, so I don't think a set is the right thing here.